### PR TITLE
nitweb: misc enhancements and cleaning

### PR DESCRIPTION
**Pull request by Morriar**
## PR summary

* remove useless modules since angular introduction
* factorize some angular code
* factorize routes /package, /group, /module... into /doc
* display error messages
* add a loading progress bar

### Unified routes

Instead of having different prefix on the kind of mentity, URLs are no unified under the `doc/` prefix:

* http://nitweb.moz-code.org/doc/core
* http://nitweb.moz-code.org/doc/core::Array
* http://nitweb.moz-code.org/doc/core::Sequence::append

### Error messages

Unknown mentities now trigger an error message:

* http://nitweb.moz-code.org/doc/core::Sequence::append2

### Loading bar

There is now an orange (but discrete) progress bar in the top of the screen while loading ressources dynamically.

Checkout the top of the screen while loading: http://nitweb.moz-code.org/

### CODE REVIEW

**Comment by Morriar** [comment]
Should fix one point of #2177

File: share/nitweb/views/doc.html [hunk 1]
```diff
@@ -0,0 +1,39 @@
+<div class='container-fluid'>
+
+	<div ng-if='error' class='alert alert-danger' role='alert'>
+		<span class='glyphicon glyphicon-exclamation-sign' aria-hidden='true'></span>
+		<span class='sr-only'>Error:</span>
+		<span ng-switch='error.status'>
+			<span ng-switch-when='404'>Entity <code>{{entityId}}</code> not found!</span>
```
Comment (jcbrinfo) [inline share/nitweb/views/doc.html]: Professional error messages do not end by an exclamation point.

* https://material.google.com/style/writing.html#writing-capitalization-punctuation
* https://material.google.com/patterns/errors.html#errors-app-errors
* https://msdn.microsoft.com/en-us/library/windows/desktop/ms679325(v=vs.85).aspx
* https://msdn.microsoft.com/en-us/library/dn742471.aspx
* http://www.openoffice.org/specs/collaterals/guides/text-style-guide.html#Exclamation
* http://stackoverflow.com/questions/3354867/does-anyone-use-a-style-guide-for-error-messages

**Comment by jcbrinfo** [comment]
:+1: +1

**Comment by privat** [comment]
+1

**Pull request merged**